src/lib/utils/os_utils.cpp

```diff
@@ -624,8 +628,22 @@ void OS::set_thread_name(std::thread& thread, const std::string& name) {
    #elif defined(BOTAN_TARGET_OS_IS_NETBSD)
    static_cast<void>(pthread_set_name_np(thread.native_handle(), "%s", const_cast<char*>(name.c_str())));
    #elif defined(BOTAN_TARGET_OS_HAS_WIN32) && defined(BOTAN_BUILD_COMPILER_IS_MSVC)
-   // Using SetThreadDescription from Win10
-   BOTAN_UNUSED(thread, name);
+   // SetThreadDescription
+   HMODULE kern = GetModuleHandleA("KernelBase.dll");
+   HRESULT(WINAPI * set_thread_name)
+   (HANDLE, PCWSTR) = reinterpret_cast<HRESULT(WINAPI*)(HANDLE, PCWSTR)>(GetProcAddress(kern, "SetThreadDescription"));
```

Comment:
A local typedef would probably make this declaration and cast a little easier on the eyes.